wg-reliability/charter.md

```diff
@@ -0,0 +1,98 @@
+# WG Reliability Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README]
+and uses the Roles and Organization Management outlined in [sig-governance].
+
+[sig-governance]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md
+[Kubernetes Charter README]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/README.md
+
+## Scope
+
+The Reliability Working Group (WG Reliability) is organized with the goal of 
+allowing users to safely use Kubernetes for managing production workloads by
+ensuring Kubernetes is stable and reliable.
+
+### In Scope
+
+- What reliability means for Kubernetes and how to measure it?
+- Measuring Kubernetes reliability in tests
+- Introducing criteria for blocking the release if the reliability is
+  below the bar
+- Building a list of end-user outages and reliability issues
+  (if applicable with mitigations and/or workarounds)
+- Creating and prioritizing a list of areas that require reliability
+  investments
+- Work with relevant SIGs on delivering necessary infrastructure
+  (e.g. test frameworks) to unblock further steps
+- Initiate and drive cross-SIG reliability improvements
+
+### Out of scope
+
+- Designing and executing improvements clearly falling into individual SIG
+  responsibilities.
+
+## Special Powers
+
+The Reliability WG has a power to block feature-oriented contributions from
+any SIG if requested reliability-related improvements are not being addressed.
```

Comment:
there will be lots of questions on the process for how this would work :)

Reply:
I described this in more details in the original doc (see Appendix):
https://docs.google.com/document/d/1IZnIG26FOJXWvW5n0Yu6zL89pH4cO8NYHKMiESc2fV0/edit

I don't think it should be part of charter, but if you think, I can add a separate doc to this PR with those information.

Reply:
I do not view the Reliability WG actually exercising this power versus recommending a project process for exercising this power.  For example, I view the moment the carrot/stick ratio is worked out, it would be a clear time to dissolve the WG.

Reply:
Once established (ie: by this WG), it feels reasonable for SIG Release to operationalize the criteria and blocking at least at a philosophical level.  Ideally all quality gates would be implemented in automation (eg: presubmit tests, conformance...not SIG Release's role to create and manage for the whole project), but I also think pragmatically and especially for completely new features it's going to take some ongoing human level analysis, human skepticism, and human level choices to reach out beyond testgrid and drive discussions on not moving a feature forward just yet and SIG Release's release team is a place that can happen.  Because test escapes are a very real thing.

For any potential implementation details though, I definitely welcome a fresh conversation around defining and implementing iteratively better feature promotion and release quality criteria.

Reply:
Hopefully addressed now. PTAL